include/experimental/__p2642_bits/layout_padded.hpp

```diff
@@ -0,0 +1,742 @@
+    }
+  }
+
+  constexpr std::array<index_type, extents_type::rank()>
+  strides() const noexcept
+  {
+    array<index_type, extents_type::rank()> __s{};
+    for (rank_type __r = 0; __r < extents_type::rank(); ++__r)
+    {
+      __s[__r] = __inner_mapping.stride(__r);
+    }
+    return __s;
+  }
+
+  constexpr index_type
+  required_span_size() const noexcept
+  {
+    return __inner_mapping.required_span_size();
+  }
+
+  /**
+   * Return the mapping given the provided indices per rank.
+   *
+   * This overload participates in overload resolution only if:
+   * - `sizeof...(Indices) == extents_type::rank()`,
+   * - `(is_convertible_v<Indices, index_type> && ...) is true`, and
+   * - (is_nothrow_constructible_v<index_type, Indices> && ...) is true.
+   */
+  MDSPAN_TEMPLATE_REQUIRES(
+    class... _Indices,
+    /* requires */ (
+      sizeof...(_Indices) == extents_type::rank()
+      && (is_convertible_v<_Indices, index_type> && ...)
+      && (is_nothrow_constructible_v<index_type, _Indices> && ...)
+    )
+  )
+  constexpr size_t operator()(_Indices... __idxs) const noexcept
+  {
+    return __inner_mapping(std::forward<_Indices>(__idxs)...);
+  }
+
+  static constexpr bool is_always_unique() noexcept { return true; }
+  static constexpr bool is_always_exhaustive() noexcept
+  {
+    return (extents_type::rank() <= size_t(1))
+      || (extents_type::static_extent(__extent_to_pad_rank) != dynamic_extent
+          && extents_type::static_extent(__extent_to_pad_rank) == __inner_extents_type::static_extent(__extent_to_pad_rank));
+  }
+  static constexpr bool is_always_strided() noexcept { return true; }
+
+  static constexpr bool is_unique() noexcept { return true; }
+  constexpr bool is_exhaustive() const noexcept
+  {
+    return (extents_type::rank() == 0)
+      || (__inner_mapping.extents().extent(__extent_to_pad_rank) == __unpadded_extent.extent(__extent_to_pad_rank));
+  }
+  static constexpr bool is_strided() noexcept { return true; }
+
+  constexpr index_type stride(rank_type __r) const noexcept
+  {
+    return __inner_mapping.stride(__r);
+  }
+
+  /**
+   * Equality operator between `layout_left_padded`s
+   *
+   * This overload only participates in overload resolution if `OtherExtents::rank() == extents_type::rank()`.
+   */
+  MDSPAN_TEMPLATE_REQUIRES(
+    class _Mapping,
+    /* requires */ (
+      detail::__is_layout_left_padded_mapping<_Mapping>::value
+      && (_Mapping::extents_type::rank() == extents_type::rank())
+    )
+  )
+  friend constexpr bool operator==(const mapping &__left, const _Mapping &__right) noexcept
+  {
+    return (__left.extents() == __right.extents()) && (!(extents_type::rank() > size_t(1)) || (__left.stride(__padding_stride_rank) == __right.stride(__padding_stride_rank)));
+  }
+
+  /**
+   * Inequality operator between `layout_left_padded`s
```

Comment:
Should we consider not enabling this in C++20 or greater mode, given that C++ will synthesize `operator!=` in that case?